webknossos/webknossos/dataset/dataset.py

```diff
@@ -750,6 +755,118 @@ def add_layer_for_existing_files(
+            ) from e
+
+        pims_images = PimsImages(
+            images,
+            timepoint=timepoint,
+            swap_xy=swap_xy,
+            flip_x=flip_x,
+            flip_y=flip_y,
+            flip_z=flip_z,
+            use_bioformats=use_bioformats,
+        )
+
+        layer = self.add_layer(
+            layer_name=layer_name,
+            category=category,
+            data_format=data_format,
+            dtype_per_channel=pims_images.dtype,
+            num_channels=pims_images.num_channels,
+        )
+        mag_view = layer.add_mag(
+            mag=mag,
+            chunk_size=chunk_size,
+            chunks_per_shard=chunks_per_shard,
+            compress=compress,
+        )
+        mag = mag_view.mag
+        layer.bounding_box = BoundingBox(
+            (0, 0, 0), pims_images.expected_shape
+        ).from_mag_to_mag1(mag)
+
+        if batch_size is None:
+            batch_size = mag_view.info.shard_size.z
+        else:
+            assert (
+                batch_size % mag_view.info.shard_size.z == 0
+            ), f"batch_size {batch_size} must be divisible by z shard-size {mag_view.info.shard_size.z}"
+
+        func_per_chunk = partial(
+            pims_images.copy_to_view,
+            mag_view=mag_view,
+        )
+
+        args = []
+        for z_start in range(0, pims_images.expected_shape.z, batch_size):
+            z_end = min(z_start + batch_size, pims_images.expected_shape.z)
+            # return shapes and set to union when using --pad
+            args.append((z_start, z_end))
+        with warnings.catch_warnings():
+            # if a slice is larger than the others it might happen that
+            # a new chunk is partially written, leading to a warning,
+            # which is ignored in this context
+            warnings.filterwarnings(
+                "ignore",
+                message=".*called on a compressed mag without block alignment.*",
+                category=RuntimeWarning,
+                module="webknossos",
+            )
+            if executor is None:
+                shapes = [func_per_chunk(i) for i in args]
+                actual_size = Vec3Int(
+                    dimwise_max(shapes) + (pims_images.expected_shape.z,)
+                )
+            else:
+                warnings.filterwarnings(
+                    "ignore",
+                    message=".*properties were found on disk which are newer than the ones that were seen last time.*",
+                    category=UserWarning,
+                    module="webknossos",
+                )
+                shapes = wait_and_ensure_success(
+                    executor.map_to_futures(func_per_chunk, args),
+                )
+                actual_size = Vec3Int(
+                    dimwise_max(shapes) + (pims_images.expected_shape.z,)
+                )
+                layer.bounding_box = BoundingBox(
+                    (0, 0, 0), actual_size
+                ).from_mag_to_mag1(mag)
```

Comment:
It could, but in the first if block the layer.bounding_box will already be correct, since every write that is larger already adapts the bounding box. Setting `layer.bounding_box` is only necessary in the distributed case, where different processes might write different bounding-boxes. I'll still put this outside the if-else to simplify the flow here, it won't do any harm.